utils/clients/tools/tests/test_isisicat_prefix_mapping.py

```diff
@@ -0,0 +1,87 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Test ICAT prefix mapping
+"""
+import unittest
+from unittest.mock import patch
+
+from utils.clients.tools.isisicat_prefix_mapping import fetch_instrument_fullname_mapping
+
+
+# pylint:disable=no-self-use,too-few-public-methods,too-many-public-methods
+class MockInstrumentQueryResult:
+    """
+    Mocks result of isisicat_prefix_mapping.client.execute_query for an instrument
+    """
+
+    def __init__(self, name, full_name):
+        self.name = name
+        self.fullName = full_name  # pylint:disable=invalid-name
+
+
+class TestICATPrefixMapping(unittest.TestCase):
+    """
+    Test ICAT prefix mapping
+    """
+
+    @patch('icat.Client.__init__', return_value=None)
+    @patch('utils.clients.tools.isisicat_prefix_mapping.AUTOREDUCTION_INSTRUMENT_NAMES',
+           ['ENGINGX'])
+    @patch('utils.clients.icat_client.ICATClient.execute_query')
+    def test_fetch_instrument_fullname_mapping_executes_icat_query(self, mock_execute_query, _):
+        """
+        Test: If instruments are fetched through a query
+        When: Testing if fetch_instrument_fullname_mapping executes an ICAT query
+        """
+        fetch_instrument_fullname_mapping()
+        mock_execute_query.assert_called_once()
+
+    @patch('icat.Client.__init__', return_value=None)
+    @patch('utils.clients.tools.isisicat_prefix_mapping.AUTOREDUCTION_INSTRUMENT_NAMES',
+           ['INVALIDINSTUMENT'])
+    @patch('logging.Logger.warning')
+    @patch('utils.clients.icat_client.ICATClient.execute_query',
+           returns_value=[MockInstrumentQueryResult("ABCDE", "ABC")])
```

Comment:
We don't need this return value here, we can just do a plain patch 
```suggestion
    @patch('utils.clients.icat_client.ICATClient.execute_query')
```

Also, I think there's a typo here. i don't think returns_value exists and should have been return_value (for this branch anyway, i think this may have actually been introduced by me 😅)